### PR TITLE
Feature/3389/search resource dialogs

### LINKED ISSUE #3389: Enable file search in resource_picker
From https://groups.google.com/forum/#!topic/refinery-cms/D2oA3ZUKFSY

> Hello,
> 
> I have changed an option in config/initializers/refinery/resources.rb that allows an admin to view more file resources other than refineries default  in the 'Add link' dialog box. 
> 
> What I would like to know is there an option in the configuration file that can order/sort file titles alphabetically then rendered/displayed. Currently files are ordered by the date they are added to refinery. Not so great if you are looking for a specific file to link to and if you have many pages. There is currently no search functionality in the add link dialog. I have attached an example add link dialog.
> 
> Thanks
> 
> Cheers Lou

We could enable search in resource_picker template like we do in image_picker: https://github.com/refinery/refinerycms/blob/master/core/app/views/refinery/admin/_resource_picker.html.erb

### PR DESCRIPTION
**Pull request by bricesanchez**
This fixes #3389

### CODE REVIEW

File: resources/app/controllers/refinery/admin/resources_controller.rb [hunk 1]
```diff
@@ -44,20 +44,24 @@ def create
...
-        render 'insert'
+
+        find_all_resources(({extra_condition[0] => extra_condition[1]} if extra_condition.present?))
+        search_all_resources if searching?
+
+        paginate_resources
+
+        @resource_area_selected = from_dialog?
+
+        render @callback.presence || 'insert'
```
Comment (bricesanchez) [inline resources/app/controllers/refinery/admin/resources_controller.rb]: Hi @parndt ! I don't find how i could render the template by context 'insert' or in '/refinery/admin/pages_dialogs/link_to' template.

Using @callback is not the good way, could you help me to find?

Comment (parndt) [reply resources/app/controllers/refinery/admin/resources_controller.rb]: oops you accidentally mentioned the `@callback` user 😉

Comment (bricesanchez) [reply resources/app/controllers/refinery/admin/resources_controller.rb]: Oops! i will not expect a callback from this user 😆

File: pages/app/controllers/refinery/admin/pages_dialogs_controller.rb [hunk 2]
```diff
@@ -28,8 +28,8 @@ def link_to
         end
 
         # web address link
-        @web_address_text = "http://"
-        @web_address_text = params[:current_link] if params[:current_link].to_s =~ /^http:\/\//
+        @web_address_text = "https://"
+        @web_address_text = params[:current_link] if params[:current_link].to_s =~ /^https:\/\//
```
Comment (parndt) [inline pages/app/controllers/refinery/admin/pages_dialogs_controller.rb]: shouldn't the `s` be optional? Like `/^https?:\/\//`

Comment (bricesanchez) [reply pages/app/controllers/refinery/admin/pages_dialogs_controller.rb]: right

File: resources/app/controllers/refinery/admin/resources_controller.rb (hunk 1 above)
Comment (bricesanchez) [reply resources/app/controllers/refinery/admin/resources_controller.rb]: Any idea @parndt ?

**Comment by parndt** [comment]
reopened to activate code climate

**Comment by bricesanchez** [comment]
Hi @parndt, could you review my PR? Also, i think the PR config of codeclimate is too strict, could we just warning us?

**Comment by parndt** [comment]
The code looks fine. I think the code climate failures provide good points 😉

**Comment by bricesanchez** [comment]
I dont't really know if we still need these lines : https://github.com/refinery/refinerycms/blob/10a6eee53686f7f51ef2a6fde6b5c2e067aa9eff/resources/app/controllers/refinery/admin/resources_controller.rb#L49-L55

I don't really know if it still works since `strong_paramaters`

**Comment by bricesanchez** [comment]
So @parndt will you acceptthis PR as it?  :)

**Comment by parndt** [comment]
Yeah. I think that the other things can be done as separate refactors. I was looking at it the other night, but it was quite intense.

**Pull request merged**